contracts/nft/src/lib.rs

```diff
@@ -0,0 +1,305 @@
+#![no_std]
+
+use gstd::{
+    collections::{HashMap, HashSet},
+    msg,
+    prelude::*,
+    ActorId,
+};
+use nft_io::*;
+
+#[derive(Debug, Default)]
+pub struct Nft {
+    pub owner_by_id: HashMap<TokenId, ActorId>,
+    pub token_approvals: HashMap<TokenId, HashSet<ActorId>>,
+    pub token_metadata_by_id: HashMap<TokenId, TokenMetadata>,
+    pub tokens_for_owner: HashMap<ActorId, Vec<TokenId>>,
+    pub token_id: TokenId,
+    pub owner: ActorId,
+    pub collection: Collection,
+    pub config: Config,
+}
+
+static mut NFT: Option<Nft> = None;
+
+#[no_mangle]
+unsafe extern fn init() {
+    let init: InitNft = msg::load().expect("Unable to decode InitNft");
+
+    let nft = Nft {
+        collection: init.collection,
+        config: init.config,
+        owner: msg::source(),
+        ..Default::default()
+    };
+    NFT = Some(nft);
+}
+
+impl Nft {
+    /// Mint a new nft using `TokenMetadata`
+    fn mint(&mut self, token_metadata: TokenMetadata) -> NftEvent {
```

Comment:
In general, in the standard it is common to indicate at which address the mint is made:
```suggestion
    fn mint(&mut self, to: &ActorId, token_metadata: TokenMetadata) -> NftEvent {
```